components/suggest/src/db.rs

```diff
@@ -731,11 +754,16 @@ impl<'a> SuggestDao<'a> {
         &mut self,
         record_id: &SuggestRecordId,
         suggestions: &[DownloadedAmpWikipediaSuggestion],
+        fields: SuggestRecord,
```

Comment:
Nit: since this is only used to determine if the record is of type `amp_mobile_suggestions`, we can just pass in a boolean `is_amp_mobile`.

```suggestion
        is_amp_mobile: bool,
```

```rs
dao.insert_amp_wikipedia_suggestions(
   record_id,
   suggestions,
   matches!(fields, SuggestRecrod::AmpMobile)
);
```

Reply:
I might actually suggest having a separate `dao.insert_amp_mobile_suggestions` method, at the risk of duplicating a bit of code!

The reason I'd suggest that is that AMP Desktop and Wikipedia suggestions are commingled in the same attachment, _but_ AMP Mobile only has mobile suggestions! So maybe we can have a method on `SuggestDao` like:

```rs
fn insert_amp_mobile_suggestions(&mut self, record_id: &SuggestRecordId, suggestions: &[DownloadedAmpSuggestion]) -> Result<()>
```

...That doesn't have to worry about Wikipedia.

Reply:
Ticket for the split: https://bugzilla.mozilla.org/show_bug.cgi?id=1879962